### ISSUE TITLE
dyld: Library not loaded error on macOS (OpenSSL missing)

### ISSUE THREAD

**Issue by kensoh** [kelaberetiv/TagUI#86]
I've come across a feedback from a macOS user that below error happens when running TagUI.

`dyld: Library not loaded: /usr/local/opt/openssl/lib/libssl.1.0.0.dylib`

This is interesting as it is the first time this is reported, not sure the difference in context why it worked for other macOS installations for other users but not for this setup. Will figure out more as I get more data points or feedback. In the meantime, trying below should be able to help. It downloads Homebrew, a package manager for macOS, installs and links OpenSSL.

```
/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
brew update && brew upgrade
brew install openssl
brew link --force openssl
```

**Comment by kensoh** [comment]
Closing this issue for time being, and keeping a lookout for more occurrences or clean macOS laptops to identify patterns for root cause.

**Comment by kensoh** [comment]
**internal reference** - this issue by other open-source project mentions that in macOS 10.13, it switches away from OpenSSL to LibreSSL - https://github.com/envoyproxy/envoy/issues/2395

**Comment by kensoh** [comment]
another datapoint the same user fixes the issue with just the following without doing the linking step (he already has Homebrew) so the first line `/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"` is not needed.

```
brew update && brew upgrade
brew install openssl
```

another datapoint where a macos user encounters the same issue. it could be the 10.13 release change in behavior around OpenSSL.

**Comment by kensoh** [comment]
This is possible a new dependency in newer version of macOS. To fix the error do the following. It installs Homebrew (a package manager for macOS) and installs OpenSSL for https connections.

**If you do not have Homebrew or don't know what is Homebrew**
```
/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
brew update && brew upgrade
brew install openssl
```

**If you already have Homebrew installed**
```
brew update && brew upgrade
brew install openssl
```

**Comment by jameschanck** [comment]
Jamess-MacBook-Pro:src jameschan$ ./tagui samples/try1 chrome
dyld: Library not loaded: /usr/local/opt/openssl/lib/libssl.1.0.0.dylib
  Referenced from: /Users/jameschan/Downloads/tagui/src/phantomjs/bin/phantomjs
  Reason: image not found

i am reading the above can fix but can explain what is wrong?? Thanks

**Comment by kensoh** [comment]
Hi @jameschanck thanks for sharing that above fix works. From my understanding macOS Sierra is switching away from OpenSSL to LibreSSL as the tool to access secure connections.

The above installs and enables OpenSSL on your macOS so that the part of TagUI (PhantomJS engine) that automates and connects to https:// websites can continue working.

**Comment by jameschanck** [comment]
Thanks. I am learning TagUI. And reading from the posts, interested to know
how to automate applications. You spoke about skuli. But not sure why u
want to integrate Roro/AHK.

**Comment by kensoh** [comment]
I see.. This [step-by-step tutorial](https://github.com/kelaberetiv/TagUI/blob/master/src/media/RPA%20Workshop.md) may be useful. Or this [video talk at FOSSASIA](https://www.youtube.com/watch?v=u1x2HOV9Jmg) with demo of TagUI features. Yes TagUI has integration with Sikuli to control UI interactions (including desktop applications) using visual recognition and OCR. But some automation scenarios needs greater control than this.

For example clicking on cell J10 on an Excel spreadsheet. Doing that by specifying a cell image to click on won't find the right cell. Open-source apps such as AHK or Roro can have automation scripts to do such actions. Integrating with them will let TagUI do more types of automation.

Now the run step can be used to call AHK or Roro, thinking if there is a way to make it even easier.

**Comment by jameschanck** [comment]
Thanks mate.
I have ceased my employment with my previous company recently.
I am thinking about providing consultancy jobs related to AI. I intend to
return to work in 6 to 12 months.
I have a request for a robot office reception.
Case illustration: a tablet at the office counter. It sense motion near it.
Wakes up and say "Hi, are you here to visit an employee or deliver?"
It listens for a name and search database & call employee to inform
visitor. In the meantime, it will communicate with visitor to register
name, contact & purpose of visit.
It listens for "deliver" too. If yes, it will trigger a drop box open, and
request the deliver person to drop the delivery. As the delivery man drop
the item, a few video will be captured as proof of delivery. And the system
will communicate with the delivery man for confirmation including generate
a signed POD for print.

I am thinking it is somewhat related to your system but require chatbot
integration, and ability to integrate with sensors.

Please share any guidance if available.,

**Comment by kensoh** [comment]
Posting as a separate issue as the discussion is not related to this macOS SSL issue - https://github.com/kelaberetiv/TagUI/issues/141

**Comment by ntoghrama** [comment]
hi! I do not have homebrew but am trying to install it for MacOS Mojave 10.14.3. 

Running:

`/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"`

Returns:

`dyld: Library not loaded: @rpath/libssl.1.0.0.dylib
  Referenced from: /Users/natasha/anaconda3/lib/libssh2.1.dylib
  Reason: image not found`


does anyone have about idea of why this does not work?

**Comment by kensoh** [comment]
Thanks for heads-up! Strange, I'm running same version of macOS - Mojave 10.14.3. Tried the same command to install on both normal terminal and Anaconda environment without issues.

`/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"`

This is probably an environment issue but can't find related issues on homebrew github or discourse. Reopening this issue for the time being to see if anyone else can add on their findings.

**Comment by kensoh** [comment]
Adding on, it looks like there is conflict with Anaconda on your system preventing installation of Homebrew, try changing PATH in ~/.profile and see if that helps - https://discourse.brew.sh/t/ssh-problem-with-install-homebrew/4052

Closing issue for now as the above installation conflict is with Anaconda / Homebrew. [Another reference](https://unix.stackexchange.com/questions/214295/osx-curl-error-with-unprivileged-user-dyld-library-not-loaded/502480#502480) on Homebrew installation issue with Anaconda.

**Comment by phosseini** [comment]
> hi! I do not have homebrew but am trying to install it for MacOS Mojave 10.14.3.
> 
> Running:
> 
> `/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"`
> 
> Returns:
> 
> `dyld: Library not loaded: @rpath/libssl.1.0.0.dylib Referenced from: /Users/natasha/anaconda3/lib/libssh2.1.dylib Reason: image not found`
> 
> does anyone have about idea of why this does not work?

If you still have the problem, this is probably your environment variable. I had the same problem and this is I solved it:

In my case, the library, libssl.1.0.0.dylib, was at this path: `/opt/local/lib` (you can go to the path and check if the file is there.) Then I added the path to my bash_profile. How to do it? First, open the bash_profile by one of the text editors like nano with this command: `nano ~/.bash_profile` and then add the following line to the bash_profile and save it: `export PATH="/opt/local/lib"` this should work. Don't forget to close the terminal and open it again after you updated the bash_profile.

**Comment by enhering** [comment]
Hi. I'm running Mac OS 10.12.6 (sierra). 

After installing mysql connector cpp 8.0.17 this same issue happened:ENH-MBP:build enhering$ ./FJH 
dyld: Library not loaded: libssl.1.0.0.dylib
  Referenced from: /usr/local/mysql-connector-c++-8.0.17/lib64/libmysqlcppconn.7.dylib
  Reason: image not found
Abort trap: 6
ENH-MBP:build enhering$ 

I already had openssl installed by brew. And I was using MySQL connector cpp 1.x.x before this without problems.

The MySQL connector cpp 8.0.17 package was downloaded from mysql website. They provide binaries for many distributions.

**Comment by kensoh** [comment]
I'm not sure what is the above command FJH that triggers the error message. Do you mean you have problems running MySQL? If yes, you should raise the issue on MySQL issues or community forum page.

If you mean an existing TagUI script now cannot work after installing MySQL, you can try the following to see if it works, by removing and installing OpenSSL again.

```
brew update && brew upgrade
brew remove openssl
brew install openssl
```

**Comment by mrtoadsc** [comment]
I stumbled across this thread while trying to resolve my own dyld: Library not loaded libssl issue. In more recent versions of MacOS, they introduced a feature called [System Integrity Protection](https://support.apple.com/en-us/HT204899) which prevents packages from writing to protected directories like /bin or /usr/lib. Even commands like `brew link openssl --force` will not work.

You must disable SIP in order to copy/link the open ssl libraries to the correct destination,  You can do this by going into Recovery Mode (reboot, Cmd-R at the Apple logo), open a Terminal and execute the command `csrutil disable`, reboot and then you can copy the files into /usr/lib. This is what worked for me. Perhaps you can copy the libssl, libcrypto files to /usr/loca/lib - which doesn't require disabling SIP - but I didn't try that.

Good luck!

**Comment by kensoh** [comment]
Thanks Mark for sharing your findings!

Adding on a short article which users can also refer to if they encounter the issue -
https://www.imore.com/how-turn-system-integrity-protection-macos

And a longer article with the technical background and screenshots of the steps -
https://www.macworld.com/article/2986118/how-to-modify-system-integrity-protection-in-el-capitan.html

**Comment by hectorb802** [comment]
> I stumbled across this thread while trying to resolve my own dyld: Library not loaded libssl issue. In more recent versions of MacOS, they introduced a feature called [System Integrity Protection](https://support.apple.com/en-us/HT204899) which prevents packages from writing to protected directories like /bin or /usr/lib. Even commands like `brew link openssl --force` will not work.
> 
> You must disable SIP in order to copy/link the open ssl libraries to the correct destination, You can do this by going into Recovery Mode (reboot, Cmd-R at the Apple logo), open a Terminal and execute the command `csrutil disable`, reboot and then you can copy the files into /usr/lib. This is what worked for me. Perhaps you can copy the libssl, libcrypto files to /usr/loca/lib - which doesn't require disabling SIP - but I didn't try that.
> 
> Good luck!

HI, I  am new in the forum,  and I am experiencing the exact same issuea after updating to MaOS Mojave,  I already Had tunred of System Integrity Protection But can you please tell me where can I get those  files that we need to copy to /usr/lib folder.. I am new using MacOS and any advice would be helpful in order to fix this issue.. 
Thanks in advance..

**Comment by kensoh** [comment]
Hi Hector, after you turned off SIP, you can restart computer and run below commands from terminal. After installation and verifying TagUI works, then restart computer to turn SIP back to the default on.

**If you do not have Homebrew or don't know what is Homebrew**
```
/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
brew update && brew upgrade
brew install openssl
```

**Or if you already have Homebrew installed**
```
brew update && brew upgrade
brew install openssl
```

**Comment by hectorb802** [comment]
> Hi Hector, after you turned off SIP, you can restart computer and run below commands from terminal. After installation and verifying TagUI works, then restart computer to turn SIP back to the default on.
> 
> **If you do not have Homebrew or don't know what is Homebrew**
> 
> ```
> /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
> brew update && brew upgrade
> brew install openssl
> ```
> 
> **Or if you already have Homebrew installed**
> 
> ```
> brew update && brew upgrade
> brew install openssl
> ```

Thanks for your answer Kensoh,

I tried your recommendation and I am still getting the same errors I have several weeks trying uninstallig and reistalling 

 I have pasted the error just in case you have a chance to check it.. Thanks again.


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/django/utils/autoreload.py", line 54, in wrapper
    fn(*args, **kwargs)
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/django/core/management/commands/runserver.py", line 109, in inner_run
    autoreload.raise_last_exception()
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/django/utils/autoreload.py", line 77, in raise_last_exception
    raise _exception[1]
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/django/core/management/__init__.py", line 337, in execute
    autoreload.check_errors(django.setup)()
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/django/utils/autoreload.py", line 54, in wrapper
    fn(*args, **kwargs)
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/django/apps/registry.py", line 114, in populate
    app_config.import_models()
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/django/apps/config.py", line 211, in import_models
    self.models_module = import_module(models_module_name)
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/django/contrib/auth/models.py", line 2, in <module>
    from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/django/contrib/auth/base_user.py", line 47, in <module>
    class AbstractBaseUser(models.Model):
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/django/db/models/base.py", line 117, in __new__
    new_class.add_to_class('_meta', Options(meta, app_label))
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/django/db/models/base.py", line 321, in add_to_class
    value.contribute_to_class(cls, name)
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/django/db/models/options.py", line 204, in contribute_to_class
    self.db_table = truncate_name(self.db_table, connection.ops.max_name_length())
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/django/db/__init__.py", line 28, in __getattr__
    return getattr(connections[DEFAULT_DB_ALIAS], item)
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/django/db/utils.py", line 201, in __getitem__
    backend = load_backend(db['ENGINE'])
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/django/db/utils.py", line 110, in load_backend
    return import_module('%s.base' % backend_name)
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/django/db/backends/postgresql/base.py", line 24, in <module>
    raise ImproperlyConfigured("Error loading psycopg2 module: %s" % e)
django.core.exceptions.ImproperlyConfigured: Error loading psycopg2 module: dlopen(/Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/psycopg2/_psycopg.cpython-37m-darwin.so, 2): Library not loaded: libssl.1.1.dylib
  Referenced from: /Users/Sindira/Documents/ezt/eztve/lib/python3.7/site-packages/psycopg2/_psycopg.cpython-37m-darwin.so
  Reason: image not found

**Comment by kensoh** [comment]
It looks like above error messages have nothing to do with TagUI. They are for a Python related Django project or psycopg2 package.

I'm afraid I can't help here, you'll have to check with the respective Python projects GitHub issues page for advice, as the solution to above error that you are facing may be different from what TagUI users are facing.

The error TagUI users are facing comes from the PhantomJS engine trying to access OpenSSL, it's a different source of error message compared to the project that you are using above, which has nothing to do with TagUI.

**Comment by AnujAroshA** [comment]
Solution give in this [link](https://github.com/kelaberetiv/TagUI/issues/86#issue-303370944) is totally correct. But if still getting the same error, try to remove existing openssl and install it again.

```
brew remove openssl
brew install openssl
```

**Comment by ntoghrama** [comment]
I finally solved this issue by removing Anaconda3 completely, and reinstalling afresh. This took care of all the dependency issues.

**Comment by ntoghrama** [comment]
I finally solved this by removing Anaconda3 completely, and reinstalling afresh. This took care of all the dependency issues.

**Comment by senadivya** [comment]
Is enable and disabling SIP required every time? I followed the suggested solution and got mongod started.  Now after the restart, my system is back with the same issues with a different reference "Referenced from: /usr/local/bin/php"

PS: Newbie

**Comment by kensoh** [comment]
For TagUI, this is only required once to install OpenSSL. From your message you mentioned `mongod`, it looks like you are troubleshooting for MongoDB daemon. Please check MongoDB github page or its support page found from the github page -

https://github.com/mongodb/mongo
https://groups.google.com/forum/#!forum/mongodb-user

**Comment by senadivya** [comment]
Basically irrespective of the action, the issue continues . Any suggestion will be of great help.
Here is the snippet:
"dyld: Library not loaded:/usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib"
dyld: Library not loaded: /usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib
  Referenced from: /usr/local/Cellar/php/7.2.10/bin/php
  Reason: image not found
Abort trap: 6

I have already completed the steps of SIP disabling and installing openssl and enabling back again.

**Comment by kensoh** [comment]
Hi Sena, can you show here the exact command you type to reproduce this error?

**Comment by kensoh** [comment]
Also, cross-linking to https://github.com/kelaberetiv/TagUI/issues/635 on an issue related to Home Brew retiring v1.0 of OpenSSL to v1.1.

**Comment by kensoh** [comment]
**Updated solution to work with newer version of Homebrew which drops OpenSSL v1.0 for v1.1**

**===== S O L U T I O N =====**

There is a new dependency in newer versions of macOS. To fix the error do the following. It installs Homebrew (a package manager for macOS) and installs OpenSSL for https connections.

**If you do not have Homebrew or don't know what is Homebrew**
```
/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
brew update && brew upgrade
brew uninstall openssl; brew uninstall openssl; brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
```

**Or if you already have Homebrew installed**
```
brew update && brew upgrade
brew uninstall openssl; brew uninstall openssl; brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
```

**===== N O T E =====**

If you encounter problems using `brew`, try the solution from @mrtoadsc here - https://github.com/kelaberetiv/TagUI/issues/86#issuecomment-532462565

**Comment by pavelgarchenko** [comment]
Here's what worked for me
1. I did what @mrtoadsc  had suggested  in https://github.com/kelaberetiv/TagUI/issues/86#issuecomment-532462565
2. I modified the solution of @kensoh https://github.com/kelaberetiv/TagUI/issues/86#issuecomment-560139852
```
brew update && brew upgrade
brew uninstall --ignore-dependencies openssl; brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
```

Hope this helps

**Comment by kensoh** [comment]
Thanks Pavel for sharing! Hate that there is so much user friction around this. @siowyisheng is now working on a POC to evaluate the possibility of overhauling TagUI to be based on NodeJS, instead of the current PhantomJS, as the execution engine.

If that is viable, the future major release would not subject users to these frictions. Also, I noticed that increasingly it becomes more friction for user and developers for macOS. Number of changes this year with macOS that need users and developers to change the way things are done and do extra. Windows does not seem to have similar issues.

**Comment by pavelgarchenko** [comment]
That's great!
You are absolutely right about dependencies. I have a dozen of other packages relying on openssl.
So, if no better solution is found, the one I've posted should be considered only after Ken's and Mark's failed tp resolve issue.

**Comment by mroderick** [comment]
I ran into an issue, possibly the same, that gave the same error message, as did https://github.com/rvm/gem-wrappers/issues/13 ... their solution was to upgrade `rvm` to latest, which also worked for me.

Perhaps it's more of an `rvm` issue?

**Comment by RuudvdvNB** [comment]
> Here's what worked for me
> 
> 1. I did what @mrtoadsc  had suggested  in [#86 (comment)](https://github.com/kelaberetiv/TagUI/issues/86#issuecomment-532462565)
> 2. I modified the solution of @kensoh [#86 (comment)](https://github.com/kelaberetiv/TagUI/issues/86#issuecomment-560139852)
> 
> ```
> brew update && brew upgrade
> brew uninstall --ignore-dependencies openssl; brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
> ```
> 
> Hope this helps
> 
> _Update:_
> Keep in mind, that I had to use `--ignore-dependencies` flag, because I have other packages installed that depend on OpenSSL.

For what it's worth, i had to use `brew intall openssl` to make it work.

**Comment by kensoh** [comment]
I assume @RuudvdvNB is referring to another tool with dependency to OpenSSL. `brew install openssl` may work for some other tools that uses OpenSSL v1.1. For TagUI and other tools relying on OpenSSL v1.0 it requires a mod, that's why I host above modified brew formula to facilitate. 

Without a modified solution, will need users to make their own builds using Xcode, which I think is too much user friction. This issue has more details on that, when Homebrew recently removed OpenSSL v1.0 and replaced with v1.1 - https://github.com/kelaberetiv/TagUI/issues/635

**Comment by MartinVincent** [comment]
FYI: In my case, the issue was stemming from a conflict betwen the OSX (10.11) ssh install and a newly installed  (via Brew) openssh client.

**Comment by brandonsturgeon** [comment]
Should I ignore warnings about extra dependencies?

i.e.
```
Error: Refusing to uninstall /usr/local/Cellar/openssl@1.1/1.1.1d
because it is required by asciinema, cairo, ffmpeg, freetds, glib, gnupg, gnutls, harfbuzz, imagemagick, libass, libevent, libheif, mysql, python, rbenv, rtmpdump, ruby, shared-mime-info, tmux, unbound, vim and w3m, which are currently installed.
You can override this and force removal with:
  brew uninstall --ignore-dependencies openssl
```

**Comment by kensoh** [comment]
It looks like you have some developer tools installed on your laptop, and that OpenSSL v1.1 is also presented, installed through Homebrew. If the reason for uninstalling v1.1 to have v1.0 in order to use TagUI, then yes you can try uninstalling to ignore the dependencies to get v1.0. However, if v1.0 has issues later on for some of those apps above, then you'll have to uninstall v1.0 to put back v1.1 for those apps to work.

Adding on, a permanent fix for this issue is being worked on now. First, PhantomJS has recently started active development, I'm checking with the maintainer on this to see if he can update to using v1.1 OpenSSL as the standard or better still, switch to LibreSSL which comes with macOS. Secondly, a POC on migrating TagUI's execution engine to Node.js is being explored to see if that makes sense for the project and its users.

**Comment by szhuzenefits** [comment]
```", line 8, in <module>
    from django.db.backends.mysql.base import DatabaseOperations as _DatabaseOperations
  File "/Users/szhu/Documents/yourPeople3/venv/lib/python2.7/site-packages/django/db/backends/mysql/base.py", line 18, in <module>
    raise ImproperlyConfigured("Error loading MySQLdb module: %s" % e)
django.core.exceptions.ImproperlyConfigured: Error loading MySQLdb module: dlopen(/Users/szhu/Documents/yourPeople3/venv/lib/python2.7/site-packages/_mysql.so, 2): Library not loaded: /usr/local/opt/openssl/lib/libssl.1.0.0.dylib
  Referenced from: /usr/local/opt/mysql/lib/libmysqlclient.20.dylib
  Reason: image not found
unable to load app 0 (mountpoint='') (callable not found or import error)
```
see this error, any suggestions? uninstall openssl and reinstall doesn't solve this problem

**Comment by kensoh** [comment]
It looks like the app you are using is still having a dependency for OpenSSL v1.0. You can try -

1. ideally, if it's possible to get an updated version of the app that uses OpenSSL v1.1
2. check with the maintainer of the project (Django?) for advice what's the right approach
3. as a last resort, try the following self-hosted formula which helps to install OpenSSL v1.0 without user having to use XCode to build for macOS (the official method from Homebrew requires user to build (details here on why the self-hosted formula is used - https://github.com/kelaberetiv/TagUI/issues/635)

```
brew uninstall openssl; brew uninstall openssl; brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
```
&ensp;
Note that above solution and this thread is to solve a problem faced by users using TagUI tool. Though it may solve similar problems faced by users of other apps, it may not. Do check the docs, issues or mailing list of the app you are using to see if they have found and recommended an official solution for their apps around this OpenSSL issue due to change in macOS (to switch to LibreSSL) and then later on a change in Homebrew (to move from OpenSSL v1.0 to v1.1).

**Comment by szhuzenefits** [comment]
> It looks like the app you are using is still having a dependency for OpenSSL v1.0. You can try -
> 
> 1. ideally, if it's possible to get an updated version of the app that uses OpenSSL v1.1
> 2. check with the maintainer of the project (Django?) for advice what's the right approach
> 3. as a last resort, try the following self-hosted formula which helps to install OpenSSL v1.0 without user having to use XCode to build for macOS (the official method from Homebrew requires user to build (details here on why the self-hosted formula is used - #635)
> 
> ```
> brew uninstall openssl; brew uninstall openssl; brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
> ```
> 
>  
> Note that above solution and this thread is to solve a problem faced by users using TagUI tool. Though it may solve similar problems faced by users of other apps, it may not. Do check the docs, issues or mailing list of the app you are using to see if they have found and recommended an official solution for their apps around this OpenSSL issue due to change in macOS (to switch to LibreSSL) and then later on a change in Homebrew (to move from OpenSSL v1.0 to v1.1).

I uninstalled the 1.1 version and installed the `v1.0.0` version fixed this issue, seems like my django need a 1.0.0 version but I have 1.1 previously, thank you!

**Comment by EvergreenTree** [comment]
I read this thread and realized that in my case it's simply caused by a recent upgrade of MacOS. In this case one only needs to open Xcode and accept Xcode's user agreement. In the case when Xcode is not installed, simply run this in terminal:
`xcode-select --install`

**Comment by emimarz** [comment]
if you install the actual version and after that copy the old version to the folder actually works
brew install // install 1.1 version
download libcrypto.1.0.0.dylib and libssl.1.0.0.dylib 
so move them to  /usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib
and /usr/local/opt/openssl/lib/libssl.1.0.0.dylib
I know that is not an elegant way but if it works is better than nothing
ah I search but for example this repository have that mentioned files
https://github.com/zoom/zoom-sdk-macos/tree/master/ZoomSDK

**Comment by frakman1** [comment]
@kensoh 

```
brew uninstall openssl; brew uninstall openssl; brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
```
Why do you uninstall openssl twice?

**Comment by kensoh** [comment]
@frakman1 the goal for this particular project is to make OpenSSL v1.0 available on macOS. Using uninstall twice is a more foolproof 1-liner to make sure any v1.1 is removed, followed by any dead and not working v1.0 is removed (which can happen and block the installation), before doing the installation from custom formula. More details in this thread - https://github.com/kelaberetiv/TagUI/issues/635#issuecomment-560185670

**Comment by kelek** [comment]
Just running `brew update && brew upgrade` worked for me;  no need to uninstall/reinstall OpenSSL.

**Comment by kensoh** [comment]
That will install OpenSSL v1.1 which is the version that Homebrew uses since its Nov release. But there are many apps which are still dependent on OpenSSL v1.0, and will not work if there is v1.1 installed instead of v1.0.

There is an official brew method to create personal tap and use extract function to install past releases, but that requires user to have XCode and build OpenSSL v1.0. So I create and host the custom formula above for users to get v1.0 without having to install and build using XCode.

**Comment by rruiz858** [comment]
> _**Welcome to [TagUI GitHub repository](https://github.com/kelaberetiv/TagUI)! TagUI is a command-line tool for digital process automation (RPA). This error is happening because macOS decided to drop OpenSSL and switched to LibreSSL. There are many other apps having this error, the solution below may or may not work for other apps, be sure to check with their respective maintainers to confirm.**_
> 
> **===== P R O B L E M =====**
> 
> I've come across a feedback from a macOS user that below error happens when running TagUI.
> 
> ```
> dyld: Library not loaded: /usr/local/opt/openssl/lib/libssl.1.0.0.dylib
> ```
> 
> **===== S O L U T I O N =====**
> 
> There is a new dependency in newer versions of macOS. To fix the error do the following. It installs Homebrew (a package manager for macOS) and installs OpenSSL for https connections.
> 
> **If you do not have Homebrew or don't know what is Homebrew**
> 
> ```
> /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
> brew update && brew upgrade
> brew uninstall openssl; brew uninstall openssl; brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
> ```
> 
>  
> **Or if you already have Homebrew installed**
> 
> ```
> brew update && brew upgrade
> brew uninstall openssl; brew uninstall openssl; brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
> ```
> 
>  
> **===== N O T E =====**
> 
> If you encounter problems using `brew`, try the solution from @mrtoadsc [near bottom of this page](https://github.com/kelaberetiv/TagUI/issues/86#issuecomment-532462565).

OMG. I love you.

**Comment by capihacendado** [comment]
Thanks so much for this solution. It saves my day in order to use mongodb commands in the terminal. 
But now, that I want to use jupyter notebook, I realized that my python is dependent of the version 1.1 

![image](https://user-images.githubusercontent.com/17687164/73341137-52c15000-427c-11ea-93c3-8b2f6ee88c71.png)

Any solution in order to not be switching among the versions?

**Comment by kensoh** [comment]
I'm afraid I have not heard of any solution to have 2 different versions of OpenSSL running on the same Mac computer. I know that sucks, especially in situations where you want to use both software at the same time but can't. Will look forward to hearing from other experts how that can be done.

**Comment by capihacendado** [comment]
> I'm afraid I have not heard of any solution to have 2 different versions of OpenSSL running on the same Mac computer. I know that sucks, especially in situations where you want to use both software at the same time but can't. Will look forward to hearing from other experts how that can be done.

thank you so much!

**Comment by LearningChanging** [comment]
> > Here's what worked for me
> > 
> > 1. I did what @mrtoadsc  had suggested  in [#86 (comment)](https://github.com/kelaberetiv/TagUI/issues/86#issuecomment-532462565)
> > 2. I modified the solution of @kensoh [#86 (comment)](https://github.com/kelaberetiv/TagUI/issues/86#issuecomment-560139852)
> > 
> > ```
> > brew update && brew upgrade
> > brew uninstall --ignore-dependencies openssl; brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
> > ```
> > 
> > 
> > Hope this helps
> > _Update:_
> > Keep in mind, that I had to use `--ignore-dependencies` flag, because I have other packages installed that depend on OpenSSL.
> 
> For what it's worth, i had to use `brew intall openssl` to make it work.

Thanks

**Comment by jjoskey** [comment]
thanks for help )

**Comment by mascardoso** [comment]
You only need to update `wget`.
`brew update && brew upgrade` is enough.

**Comment by tonycodes** [comment]
For me i had to run `brew reinstall php` and i'm back up and running

**Comment by benhorns** [comment]
> > _**Welcome to [TagUI GitHub repository](https://github.com/kelaberetiv/TagUI)! TagUI is a command-line tool for digital process automation (RPA). This error is happening because macOS decided to drop OpenSSL and switched to LibreSSL. There are many other apps having this error, the solution below may or may not work for other apps, be sure to check with their respective maintainers to confirm.**_
> > **===== P R O B L E M =====**
> > I've come across a feedback from a macOS user that below error happens when running TagUI.
> > ```
> > dyld: Library not loaded: /usr/local/opt/openssl/lib/libssl.1.0.0.dylib
> > ```
> > 
> > 
> > **===== S O L U T I O N =====**
> > There is a new dependency in newer versions of macOS. To fix the error do the following. It installs Homebrew (a package manager for macOS) and installs OpenSSL for https connections.
> > **If you do not have Homebrew or don't know what is Homebrew**
> > ```
> > /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
> > brew update && brew upgrade
> > brew uninstall openssl; brew uninstall openssl; brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
> > ```
> > 
> > 
> >  
> > **Or if you already have Homebrew installed**
> > ```
> > brew update && brew upgrade
> > brew uninstall openssl; brew uninstall openssl; brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
> > ```
> > 
> > 
> >  
> > **===== N O T E =====**
> > If you encounter problems using `brew`, try the solution from @mrtoadsc [near bottom of this page](https://github.com/kelaberetiv/TagUI/issues/86#issuecomment-532462565).
> 
> OMG. I love you.

> I stumbled across this thread while trying to resolve my own dyld: Library not loaded libssl issue. In more recent versions of MacOS, they introduced a feature called [System Integrity Protection](https://support.apple.com/en-us/HT204899) which prevents packages from writing to protected directories like /bin or /usr/lib. Even commands like `brew link openssl --force` will not work.
> 
> You must disable SIP in order to copy/link the open ssl libraries to the correct destination, You can do this by going into Recovery Mode (reboot, Cmd-R at the Apple logo), open a Terminal and execute the command `csrutil disable`, reboot and then you can copy the files into /usr/lib. This is what worked for me. Perhaps you can copy the libssl, libcrypto files to /usr/loca/lib - which doesn't require disabling SIP - but I didn't try that.
> 
> Good luck!
I have tried all of the above and none of this is working for me :(((

I am trying to do the installation detailed here: https://www.pyimagesearch.com/2018/08/17/install-opencv-4-on-macos/

when I submit this command in terminal: sudo python3 get-pip.py

I get this error:  

WARNING: The directory '/Users/aislinnmcnamara/Library/Caches/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
WARNING: pip is configured with locations that require TLS/SSL, however the ssl module in Python is not available.
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.",)': /simple/pip/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.",)': /simple/pip/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.",)': /simple/pip/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.",)': /simple/pip/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.",)': /simple/pip/
Could not fetch URL https://pypi.org/simple/pip/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pip/ (Caused by SSLError("Can't connect to HTTPS URL because the SSL module is not available.",)) - skipping
ERROR: Could not find a version that satisfies the requirement pip (from versions: none)
ERROR: No matching distribution found for pip
WARNING: pip is configured with locations that require TLS/SSL, however the ssl module in Python is not available.
Could not fetch URL https://pypi.org/simple/pip/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pip/ (Caused by SSLError("Can't connect to HTTPS URL because the SSL module is not available.",)) - skipping


anyone please help!?

**Comment by kensoh** [comment]
I'm afraid issue in above post has nothing to do with this thread. It looks like you are having problem installing `pip`. Try to see if you can install Python from here https://www.python.org/downloads/ as Python would already come installed with pip, no need to do installation separately from command line. If that doesn't work, you can raise an issue at Python's PIP repo page to see if anyone has encountered this problem and has a solution - https://github.com/pypa/pip

**Comment by kelek** [comment]
I've had brew for some odd reason mysteriously uninstall pip.  You can try
```
sudo /usr/bin/easy_install pip
```
or
```
curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
/usr/bin/python get-pip.py
```
to reinstall it.

**Comment by capihacendado** [comment]
> I'm afraid I have not heard of any solution to have 2 different versions of OpenSSL running on the same Mac computer. I know that sucks, especially in situations where you want to use both software at the same time but can't. Will look forward to hearing from other experts how that can be done.

After getting tired of being switching among versions, I have decided to look for one solution to my situation. I wanted to reinstall mongodb in order to see if mongo was truly dependent of openssl 1.0.2 or it was just my version of mongodb. I was surprised when I found that Homebrew has delete mongodb from its core formulas since mongo was migrated to a non open-source license. 
So, following this thread in stackOverflow I was able to install mongodb again using tap https://stackoverflow.com/questions/57856809/installing-mongodb-with-homebrew
So I run this command in my terminal 

```
brew tap mongodb/brew
brew install mongodb-community
brew services start mongodb-community
```

and voila, I can used mongo's terminal commands with openssl 1.1

**Comment by hydrosquall** [comment]
I resolved the issue by switching my active OpenSSL version after installing the custom tap provided above

```
brew switch openssl 1.0.2t
```

Source:

**Comment by philippcheung** [comment]
brew switch openssl 1.0.2s

https://stackoverflow.com/a/59184347

**Comment by mnooel** [comment]
To correct this I tried to uninstall freetds and unixodbc to then reinstall. I uninstalled the first version of freetds but when I tried to unistall unixodbc I got the following error:

`Error: Refusing to uninstall /usr/local/Cellar/unixodbc/2.3.7
because it is required by freetds@91.0, which is currently installed.
You can override this and force removal with:
  brew uninstall --ignore-dependencies unixodbc`

It seems like I had two versions of freetds on my computer. So I tried unistalling the specific version and then unixodbc:

`brew uninstall freetds@91.0 ; brew uninstall unixodbc`
`brew install unixodbc; brew install freetds`

and it worked

**Comment by johnkjoseph** [comment]
@capihacendado

Since your error shows it is looking for files in openssl@1.1 folders, you can try installing brew install openssl@1.1, navigate to` /usr/local/Cellar/` and copy the folder. Then do brew reinstall `https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb`. Go back to  `/usr/local/Cellar/` place the copied folder openssl@1.1 and make sure symlink to openssl@1.1 in `usr/local/opt` as openssl@1.1. Obviously a hack but it worked for me while trying to set up PHP 5.6 where it was complaining about OpenSSL v1.0.0 and OpenSSL v1.1

**Comment by JivanRoquet** [comment]
This command
```
brew upgrade
```
can be a massive disrupter for your entire system and shouldn't be taken (nor recommended) lightheartedly. It will upgrade every single package from your installation, which is something you don't necessarily want.

Recommending this with no warnings just to fix an openssl package is quite irresponsible, and is akin to burning an entire forest for someone who just wants to warm his dinner.

**Comment by kensoh** [comment]
Hi @JivanRoquet thanks for pointing out! I have the line `brew update && brew upgrade` as part of the solution when I originally research for the solution to the problem. Though this was way back and I can't backtrace the source of having this command.

The solution works for me and I believe works for significant number of people, thus the upvotes and positive reinforcement that this is still the best solution. From your feedback, I can understand why this line could be a big problem, and I've removed that line from the solution summary.

Hopefully the solution still works without that line!

It theoretically should, because the solution has changed along the way with Homebrew migration to OpenSSL v1.1, to require hosting a custom formula in order to solve the problem without XCode dependency on user laptops.

**Comment by sinaahmadi** [comment]
Just upgraded to `macOS Catalina`, I run into this same problem. I preferred not to modify anything regarding my System Integrity Protection, so the following could fix my problem:

```
brew update && brew upgrade
brew uninstall --ignore-dependencies openssl
brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
```

**Comment by kolja-lucht** [comment]
> @kensoh this command
> 
> ```
> brew upgrade
> ```
> 
> can be a massive disrupter for your entire system with no way back, and shouldn't be taken (nor recommended) lightheartedly. Depending on your system, it will get your computer busy for hours, and will upgrade every single package from your installation, which is something you probably (most certainly) do NOT want.
> 
> Recommending this with no warnings just to fix an openssl package is quite irresponsible, and is akin to burning an entire forest for someone who just wants to warm his dinner. I can't believe this post was so overwhelmingly upvoted.

@JivanRoquet thanks for pointing that out! Running `brew update $ brew upgrade` got me in this situation in the first place. While looking for a fix I came across this thread. What worked for me were two of the lines from @sinaahmadi 
```
brew uninstall --ignore-dependencies openssl
brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
```
Thanks! 😊

**Comment by kensoh** [comment]
Adding on, below flag will ensure uninstallation of openssl, but I'm won't and can't put this into the recommended solution at top of the page, at least for TagUI tool users.

```
brew uninstall --ignore-dependencies openssl
```

This is because the flag will force uninstallation even though there are other apps on the computer that requires the openssl version that is being uninstalled. While that may fix the problem for one app, it almost certainly will break some other app(s) on the Mac laptop.

For eg, it could make TagUI or the tool you are fixing to work, while breaking other apps that are dependent on a particular version of openssl. Will have to leave it to users to decide and make a call what is more important to be working.

**Comment by marcelloinfoweb** [comment]
> Just upgraded to `macOS Catalina`, I run into this same problem. I preferred not to modify anything regarding my System Integrity Protection, so the following could fix my problem:
> 
> ```
> brew update && brew upgrade
> brew uninstall --ignore-dependencies openssl
> brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
> ```

After this procedure, this message appears

dyld: Library not loaded: /usr/local/opt/icu4c/lib/libicui18n.64.dylib
Referenced from: /usr/local/opt/php@5.6/sbin/php-fpm
Reason: image not found

MacOs Catalina
PHP 5.6

**Comment by kensoh** [comment]
@marcelloinfoweb this looks like an unusual error message. macOS Catalina migrated to PHP v7.1, not sure why PHP 5.6 is still being used on your system. Do you mean you are getting above error when you run `tagui` command on your macOS computer?

**Comment by heatherbooker** [comment]
@kensoh Thanks for all your work debugging in this thread!

https://github.com/kelaberetiv/TagUI/issues/86#issuecomment-371766145 was the first thing I tried and it would be awesome to have JivanRoquet's disclaimer that one should be careful when running those lines added to that message, if you would consider that. :)

**Comment by kensoh** [comment]
Thanks @heatherbooker @kolja-lucht and @JivanRoquet who originally pointed this out.

I agree that something should be done. This thread has 4000 unique visitors EVERY WEEK.

Though the official solution at the top is updated to remove this risky command, there are bits and pieces scattered in the whole thread which still has the `brew upadate && brew upgrade` command.

In order to avoid collateral damage by any visitors, I have decided and removed all references of this command, including posts made by users. Though this modification will somehow warp the conversation thread of how the solution developed, and may break the flow of what other users have shared.

It is still a smaller cost than if anyone accidentally run that supposed best practice command and got into issues with the dev environment. I found that command somewhere as the recommended solution but it was too long ago and probably the source has been updated.

Where the reference can't be removed legibly, I over-wrote with below code block.

```
# outdated solution in this code block, see updated solution above
https://github.com/kelaberetiv/TagUI/issues/86#issue-303370944
```

**Comment by Adsidera** [comment]
I have a similar error after the same procedure quoted by @marcelloinfoweb , but with this error message: 

`Library not loaded: /usr/local/opt/openssl@1.1/lib/libssl.1.1.dylib`

**Comment by imhamad** [comment]
Same issue

**Comment by narkan** [comment]
Really depressing - the disasterous brew upgrade command has wrecked everything! I can't even remember what problem I was trying to solve when I read that "solution"... now in a nightmare of variations on the following, no matter which advice I follow from above:
```
dyld: Library not loaded: /usr/local/opt/openssl@1.1/lib/libcrypto.1.1.dylib
  Referenced from: /usr/local/bin/php
  Reason: image not found
```

I use MAMP Pro on Mac Catalina - not sure if that and it's diferent installation of php are hiding the problem...

**Comment by ernesthernandez** [comment]
> Really depressing - the disasterous brew upgrade command has wrecked everything! I can't even remember what problem I was trying to solve when I read that "solution"... now in a nightmare of variations on the following, no matter which advice I follow from above:
> 
> ```
> dyld: Library not loaded: /usr/local/opt/openssl@1.1/lib/libcrypto.1.1.dylib
>   Referenced from: /usr/local/bin/php
>   Reason: image not found
> ```
> 
> I use MAMP Pro on Mac Catalina - not sure if that and it's diferent installation of php are hiding the problem... I also suspect that in my eagerness to get it fixed, running brew uninstall **--ignore-dependencies** openssl was probably stupid, as now a load of other programs have lost their dependencies... grrrrr.
> 
> Any other suggestions appreciated - I don't want to disable SIP. Must be a better solution...!

`brew reinstall openssl@1.1`

**Comment by narkan** [comment]
> `brew reinstall openssl@1.1`

Brilliant - that solved one of the issues - thank you!!

But I still get the following, where it's now looking in the **/openssl/** rather than /openssl@1.1/ directory:

```
dyld: Library not loaded: /usr/local/opt/openssl/lib/libssl.1.0.0.dylib
  Referenced from: /usr/local/opt/mysql@5.7/bin/mysql
  Reason: image not found
```
Is there a way to install an unversioned version?? If I try **brew reinstall openssl** it says: **Reinstalling openssl@1.1**

**Comment by ernesthernandez** [comment]
> > `brew reinstall openssl@1.1`
> 
> Brilliant - that solved one of the issues - thank you!!
> 
> But I still get the following, where it's now looking in the **/openssl/** rather than /openssl@1.1/ directory:
> 
> ```
> dyld: Library not loaded: /usr/local/opt/openssl/lib/libssl.1.0.0.dylib
>   Referenced from: /usr/local/opt/mysql@5.7/bin/mysql
>   Reason: image not found
> ```
> 
> Is there a way to install an unversioned version?? If I try **brew reinstall openssl** it says: **Reinstalling openssl@1.1**

you could try
`brew uninstall openssl; brew uninstall openssl; brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb`

**Comment by lionelyoung** [comment]
Switch to an older openssl package, via SO

`brew switch openssl 1.0.2s`

**Comment by razvanphp** [comment]
@lionelyoung did that and now I have this:

```
dyld: Library not loaded: /usr/local/opt/icu4c/lib/libicui18n.64.dylib
  Referenced from: /usr/local/bin/php
  Reason: image not found
Abort trap: 6
```

**Comment by nandbarkin** [comment]
The resolved for me after upgrade wget to latest version on mac OS version: 10.14.6 (18G4032)
```
wget -V
GNU Wget 1.20.3 built on darwin18.7.0.

-cares +digest -gpgme +https +ipv6 +iri +large-file -metalink +nls
+ntlm +opie -psl +ssl/openssl

Wgetrc:
    /usr/local/etc/wgetrc (system)
Locale:
    /usr/local/Cellar/wget/1.20.3_2/share/locale
Compile:
    clang -DHAVE_CONFIG_H -DSYSTEM_WGETRC="/usr/local/etc/wgetrc"
    -DLOCALEDIR="/usr/local/Cellar/wget/1.20.3_2/share/locale" -I.
    -I../lib -I../lib -I/usr/local/opt/openssl@1.1/include -DNDEBUG -g
    -O2
Link:
    clang -DNDEBUG -g -O2 -lidn2 -L/usr/local/opt/openssl@1.1/lib -lssl
    -lcrypto -ldl -lz ftp-opie.o openssl.o http-ntlm.o ../lib/libgnu.a
    -liconv -lintl -Wl,-framework -Wl,CoreFoundation -lunistring

Copyright (C) 2015 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later
<http://www.gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Originally written by Hrvoje Niksic <hniksic@xemacs.org>.
Please send bug reports and questions to <bug-wget@gnu.org>.
```

**Comment by snusmu** [comment]
found this link since my wget was not working, for wget reinstalling it solved the issue on 10.15.4

```
brew uninstall wget
brew install wget
```

**Comment by iamshreeram** [comment]
@razvanphp / @snusmu -
If it is mac, You can find the version of openssl from here - `ls /usr/local/Cellar/openssl`. 

Based on version you find, you can switch your openssl version :

```
brew switch openssl 1.0.2r
# or 
brew switch openssl 1.0.2s
# or
brew switch openssl 1.0.2t
```

This will resolve the wget issue.

**Comment by fusion27** [comment]
> found this link since my wget was not working, for wget reinstalling it solved the issue on 10.15.4
> 
> ```
> brew uninstall wget
> brew install wget
> ```

Per @snusmu I solved with an uninstall/re-install.  I also removed wget customizations for a fresh install
```sh
brew uninstall wget && rm -rf /usr/local/etc/wgetrc && brew install wget
```

**Comment by ns11** [comment]
Just do a `brew switch openssl 1.0.2t`, in my case the only openssl 1.0.x version previously installed by with brew

**Comment by narkan** [comment]
```
brew switch openssl 1.0.2t
```
Thank you ns11! That fixed it for me!! 👍

**Comment by rsha256** [comment]
I've been having this same issue with wget, tried reinstalling openssl to no avail.

```sh
brew uninstall wget
brew install wget

```
^this finally made it work for me 🙌

**Comment by huy-springlabs** [comment]
```
brew switch openssl 1.0.2t
```

After half a day of googling of attempting to install/reinstall openssl, this finally worked for me. Thanks

**Comment by n0es1s** [comment]
brew switch openssl 1.0.2r worked for me as 1.0.2t was showing:  openssl does not have a version "1.0.2t" in the Cellar.

**Comment by agermain** [comment]
+1 to `brew switch openssl 1.0.2t`

**Comment by DuckMasterAl** [comment]
The proper brew command should be ```brew uninstall --ignore-dependencies openssl;``` since openssl will probably be in some dependencies (at least it was for me)

**Comment by abodrak** [comment]
after applying this solution, I can't use pip anymore , pip cannot use https connections to download packages, 
is there any other than this solution (downgrading OpenSSL) ?

**Comment by juanitourquiza** [comment]
The perfect solution in macOS Catalina with PHP 7.2
https://github.com/kelaberetiv/TagUI/issues/86#issue-303370944

**Comment by doronhorwitz** [comment]
I tried @kensoh recommendation
```
brew uninstall openssl; brew uninstall openssl; brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
```
to no avail.

The thing that worked for me

**Comment by iamshreeram** [comment]
This is a single-line command that will fix this issue - 

```
brew switch openssl `ls /usr/local/cellar/openssl | sed 's/\///g'
```

**Comment by iamshreeram** [comment]
This is a single-line command that will fix this issue -
```
brew switch openssl `ls /usr/local/cellar/openssl | sed 's/\///g'`
```

**Comment by kuzeko** [comment]
As @DuckMasterAl  said, the correct command is:

```
brew uninstall --ignore-dependencies openssl; brew uninstall --ignore-dependencies openssl; brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
```

**Comment by raulflags** [comment]
> **Welcome to [TagUI GitHub repository](https://github.com/kelaberetiv/TagUI) 🤖! TagUI is a command-line tool for RPA (to automate digital processes). There's a Python package built on TagUI called [RPA for Python](https://github.com/tebelorg/RPA-Python) 🐍 with a similar issue, but it's automatically fixed with below solution during initialisation.**
> 
> **This error is happening because macOS decided to drop OpenSSL and switched to LibreSSL. Furthermore, macOS Homebrew switched from OpenSSL v.1.0 to v1.1, breaking many other apps that are dependent on OpenSSL v1.0. The solution below may or may not work for other apps, be sure to check with their respective maintainers to confirm.**
> 
> _**===== P R O B L E M =====**_
> 
> I've come across a feedback from a macOS user that below error happens when running TagUI.
> 
> ```
> dyld: Library not loaded: /usr/local/opt/openssl/lib/libssl.1.0.0.dylib
> ```
> 
> _**===== S O L U T I O N =====**_
> 
> There is a new dependency in newer versions of macOS. To fix the error do the following. It installs Homebrew (a package manager for macOS) and installs OpenSSL for https connections. There are 2 uninstall commands below to first remove OpenSSL v1.1, then remove any dead OpenSSL v1.0, before installing a custom OpenSSL v1.0 formula which works without having to build with Xcode.
> 
> **If you don't have Homebrew or don't know what is it**
> 
> ```
> /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
> brew uninstall openssl; brew uninstall openssl; brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
> ```
> 
>  
> **Or if you already have Homebrew installed**
> 
> ```
> brew uninstall openssl; brew uninstall openssl; brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb
> ```
> 
>  
> _**===== O N E  L A S T  T H I N G . . . =====**_
> 
> If you encounter problems using `brew`, try the solution from @mrtoadsc [near bottom of this page](https://github.com/kelaberetiv/TagUI/issues/86#issuecomment-532462565).

Its Sept/2020 and this still works smoothly!! Thanks! :D

**Comment by PoslinskiNet** [comment]
After upgrades to Catalina: 

`brew install --force https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb` 

helped in my case (--force flag did the trick)

**Comment by Stwissel** [comment]
I had that problem too today on macOS 10.14.6
The solution was: **don't touch brew** - you don't want OpenSSL 1.0 on your system
Head over to the phantomJS [download page](https://phantomjs.org/download.html) and download the last `release 2.1.1`
Unzip the the download and locate the `phantomjs` file (in bin) replace the file in `[taguihome]/src/phantomjs/bin` with it. This will work with OpenSSL 1.1.1 (what's current in brew)

**Comment by sammoh94** [comment]
I followed the instructions outlined [here](https://github.com/kelaberetiv/TagUI/issues/86#issuecomment-532462565) and disabled System Integrity Protection. When I try to run these steps: `brew uninstall openssl; brew uninstall openssl; brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb`, I still get the following error:

```
Calling Non-checksummed download of openssl formula file from an arbitrary URL is disabled! Use 'brew extract' or 'brew create' and 'brew tap-new' to create a formula file in a tap on GitHub instead.
```

Any suggestions?

**Comment by Stwissel** [comment]
> Any suggestions?
Don’t bother with this approach. While SIP is a pita, it is there for a reason. Also you don’t want OpenSSL1.0 on your system. 
Follow the steps I outlined above and replace the phantomJS with a newer version.

**Comment by KungFuLucky7** [comment]
This worked for me:
```bash
brew install rbenv/tap/openssl@1.0
ln -sfn /usr/local/Cellar/openssl@1.0/1.0.2t  /usr/local/opt/openssl
```

**Comment by Maks1708** [comment]
Hi!

Setup all. I want to run an example sudo src/tagui flows/samples/2_github. tag

Chrome returns this error.

"Connection is not protected
Hackers may try to steal your data from the site www.google.com (for example, passwords, messages, or Bank card numbers). Learn more...
NET:: ERR_CERT_INVALID"

How to fix it?

Thanks!
<img width="1279" alt="Screenshot at Sep 18 01-09-00" src="https://user-images.githubusercontent.com/11895267/93533536-d46d0780-f94b-11ea-8524-ff25097a1d75.png">

**Comment by Stwissel** [comment]
@Maks1708 -> 
Head over to the phantomJS download page and download the last release 2.1.1
Unzip the the download and locate the phantomjs file (in bin) replace the file in [taguihome]/src/phantomjs/bin with it. This will work with OpenSSL 1.1.1 (what's current in brew)

**Comment by Maks1708** [comment]
> @Maks1708 ->
> Head over to the phantomJS download page and download the last release 2.1.1
> Unzip the the download and locate the phantomjs file (in bin) replace the file in [taguihome]/src/phantomjs/bin with it. This will work with OpenSSL 1.1.1 (what's current in brew)

Thank you @Stwissel !

I changed phantomJS and switch brew to openssl 1.1.1g. But chrome gives the same error.

**Comment by Maks1708** [comment]
> @Maks1708 ->
> Head over to the phantomJS download page and download the last release 2.1.1
> Unzip the the download and locate the phantomjs file (in bin) replace the file in [taguihome]/src/phantomjs/bin with it. This will work with OpenSSL 1.1.1 (what's current in brew)

I opened tagui in a text editor and fixed this block. Added flags: --ignore-certificate-errors --ignore-urlfetcher-cert-requests

Started

```
if ! type "$chrome_command" > /dev/null; then
echo "ERROR - cannot find Chrome command \"$chrome_command\""
echo "update chrome_command setting in tagui/src/tagui and make sure symlink to command is created"; exit 1; fi
chrome_process_id="$(ps x | grep remote-debugging-port=9222 | grep tagui_user_profile | grep window-size | sed -e 's/^[ ]*//' | cut -d' ' -f 1 | sort -nur | head -n 1)"
if [ -n "$chrome_process_id" ]; then kill $chrome_process_id; fi
$chrome_command $chrome_switches $window_size $headless_switch --ignore-certificate-errors --ignore-urlfetcher-cert-requests > /dev/null 2>&1 &
fi
```

**Comment by jzaburunov** [comment]
> ... Perhaps you can copy the libssl, libcrypto files to /usr/loca/lib - which doesn't require disabling SIP - but I didn't try that.

worked for me

**Comment by chengsam** [comment]
> This worked for me:
> 
> ```shell
> brew install rbenv/tap/openssl@1.0
> ln -sfn /usr/local/Cellar/openssl@1.0/1.0.2t /usr/local/opt/openssl
> ```

This works for me!

**Comment by juanfung** [comment]
@KungFuLucky7's solution worked for me on Catalina and Homebrew 2.5.2

**Comment by londerson** [comment]
1- Download the file:
`wget https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb`

2 - Run brew with the file downloaded:
`brew install openssl.rb`

**Comment by michaelcaterisano** [comment]
> 1- Download the file:
> `wget https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb`
> 
> 2 - Run brew with the file downloaded:
> `brew install openssl.rb`

Works for me!

**Comment by rodriguezcoding** [comment]
@londerson thanks!

**Comment by erinoggz** [comment]
> brew install https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb

im getting this error 
Error: Calling Non-checksummed download of openssl formula file from an arbitrary URL is disabled! Use 'brew extract' or 'brew create' and 'brew tap-new' to create a formula file in a tap on GitHub instead.
If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
  https://github.com/tebelorg/Tump/issues/new

**Comment by chelseatroy** [comment]
> This worked for me:
> 
> ```shell
> brew install rbenv/tap/openssl@1.0
> ln -sfn /usr/local/Cellar/openssl@1.0/1.0.2t /usr/local/opt/openssl
> ```

Thank you @KungFuLucky7. You are a gentleman and a scholar! Worked for me on Catalina and Homebrew 2.5.8

**Comment by gauravkeerthi** [comment]
I agree with the earlier commenter. Please DO NOT INSTALL OPENSSL. Just update your phantomjs.
https://github.com/kelaberetiv/TagUI/issues/86#issuecomment-689781003

**Comment by xlcrr** [comment]
`brew update` 

gave me 2 commands to run that fixed it 

```
git -C /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core fetch --unshallow
git -C /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cast fetch --unshallow
```

**Comment by kensoh** [comment]
Hi Guys, for TagUI @Stwissel may have found the permanent solution. Note that this solution may not help for other projects.
Download directly from below link. After unzipping, put contents of unzipped folder into tagui/src/phantomjs folder.

https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-2.1.1-macosx.zip

Though above zip is PhantomJS v2.1.1 (same as TagUI packaged PJS), total file size for this zip after extraction is different. It might be Ariya, creator of PhantomJS, put in a fix already just to solve this headache with macOS/OpenSSL/Homebrew.

**Comment by sparsh9** [comment]
.rbenv/versions/3.0.2/bin/ruby is loading libcrypto in an unsafe way
zsh: abort      rails db:create

@kensoh , Im getting this error on MacBook Pro, intel 2017, and this solution is not working, can you help?

**Comment by kensoh** [comment]
Tagging @ruthtoh as I no longer maintains this TagUI repo

**Comment by amitboss2015** [comment]
I have also tried to install openssl 1.0.0 but not successful.
machine config - 2.6 GHz 6-Core Intel Core i7
macos - ventura - 3.5.1 

brew install rbenv/tap/openssl@1.0
*************** ~ % brew install rbenv/tap/openssl@1.0
==> Fetching rbenv/tap/openssl@1.0
==> Downloading https://www.openssl.org/source/openssl-1.0.2t.tar.gz
Already downloaded: *****************--openssl-1.0.2t.tar.gz
==> Installing openssl@1.0 from rbenv/tap
==> perl ./Configure --prefix=/usr/local/Cellar/openssl@1.0/1.0.2t --openssldir=/usr/local/etc/openssl no-ssl2 no-ssl3 no-zlib shared enable-cms darwin64-x86_64-cc enable-ec_nistp_64_gcc_128
==> make depend
==> make
==> make test
Last 15 lines from /Users/*****/Library/Logs/Homebrew/openssl@1.0/04.make:
140704590104492:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994:
140704590104492:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994:
140704590104492:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994:
140704590104492:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994:
140704590104492:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994:
140704590104492:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994:
140704590104492:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994:
140704590104492:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994:
140704590104492:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994:
140704590104492:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994:
140704590104492:error:1007C06B:elliptic curve routines:EC_POINT_set_affine_coordinates_GFp:point is not on curve:ec_lib.c:968:
140704590104492:error:1007C06B:elliptic curve routines:EC_POINT_set_affine_coordinates_GFp:point is not on curve:ec_lib.c:968:
140704590104492:error:1007C06B:elliptic curve routines:EC_POINT_set_affine_coordinates_GFp:point is not on curve:ec_lib.c:968:
make[1]: *** [test_ec] Error 1
make: *** [tests] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/homebrew-core):
  https://github.com/rbenv/homebrew-tap/issues

These open issues may also help:
`brew install rbenv/tap/openssl@1.0` not working anymore https://github.com/rbenv/homebrew-tap/issues/1
Error when installing openssl version 1.0 (MacOS Big Sur, Version 11.7.6. i7) https://github.com/rbenv/homebrew-tap/issues/9

**Comment by ladar** [comment]
This issue is closed, so not sure if the bug is fixed alreadt, but I noticed a couple of other related issues. so posting anyways. Check out my explanation [here](https://github.com/sidneys/homebrew-homebrew/issues/23): 

The certs embedded in the tarball and used by the unit tests expired.  You can find patches for 1.02 and 1.1.1 here:

[1.1.1_update_expiring_certificates.patch](https://raw.githubusercontent.com/lavabit/magma/682101e08114be9b006aadb228943c487dfb1abf/lib/patches/openssl/1.1.1_update_expiring_certificates.patch)

[1.0.2_update_expiring_certificates.patch](https://raw.githubusercontent.com/lavabit/magma/682101e08114be9b006aadb228943c487dfb1abf/lib/patches/openssl/1.0.2_update_expiring_certificates.patch)

I only tweaked the alignment of the 1.1.1 patch so it would line up and apply to 1.0.2 without complaint. I would post a link to the original 1.1.1 patch but I don't have it handy.

All these patches do is update the shell script used to make those test certs. The developer ran the updated script and generated new dummy test certs, and the remaining bitsare used to  update the certs themselves.

**Comment by Gamma-Software** [comment]
@magicliang  +1

**Comment by Gamma-Software** [comment]
@magicliang  I installed python with brew with the command `brew install python@3.10` for python 3.10 and it works now
